src/index.ts

```diff
@@ -197,23 +197,37 @@ export class IDX {
     }
   }
 
-  async _getIDXDoc(did: string): Promise<Doctype> {
-    const doc = await this._ceramic.createDocument(
+  async _createIDXDoc(did: string): Promise<Doctype> {
+    return await this._ceramic.createDocument(
       'tile',
       {
         deterministic: true,
-        metadata: { controllers: [did], family: 'IDX', schema: schemas.IdentityIndex },
+        metadata: { controllers: [did], family: 'IDX' },
       },
       { anchor: false, publish: false }
     )
+  }
+
+  async _getIDXDoc(did: string): Promise<Doctype> {
+    const doc = await this._createIDXDoc(did)
     if (doc.metadata.schema !== schemas.IdentityIndex) {
       throw new Error('Invalid document: schema is not IdentityIndex')
     }
     return doc
   }
 
   async _getOwnIDXDoc(): Promise<Doctype> {
-    return await this._getIDXDoc(this.id)
+    const doc = await this._createIDXDoc(this.id)
+    if (doc.metadata.schema == null) {
+      // Doc just got created, need to update it with schema
+      await doc.change(
+        { metadata: { ...doc.metadata, schema: schemas.IdentityIndex } },
+        { anchor: true, publish: true }
```

Comment:
These are default. Don't think you need to set explicitly.

Reply:
Good, to know, thanks!